File: frontend/src/types/layoutZone.ts
// backend/app/schemas/layout_zone.py ile birebir eşleşir.
export interface LayoutZoneProduct {
  id: number;
  name: string;
}

export interface LayoutZoneOut {
  id: number;
  name: string;
  x: number;
  y: number;
  width: number;
  height: number;
  products: LayoutZoneProduct[];
}

export interface LayoutZoneCreatePayload {
  name: string;
  width: number;
  height: number;
}

export interface LayoutZoneUpdatePayload {
  name?: string;
  width?: number;
  height?: number;
  x?: number;
  y?: number;
}

File: backend/app/schemas/layout_zone.py
from pydantic import BaseModel, Field


class LayoutZoneProduct(BaseModel):
    id: int
    name: str


class LayoutZoneCreate(BaseModel):
    name: str = Field(max_length=100)
    width: int = Field(gt=0)
    height: int = Field(gt=0)


class LayoutZoneUpdate(BaseModel):
    name: str | None = Field(default=None, max_length=100)
    width: int | None = Field(default=None, gt=0)
    height: int | None = Field(default=None, gt=0)
    x: int | None = None
    y: int | None = None


class LayoutZoneOut(BaseModel):
    id: int
    name: str
    x: int
    y: int
    width: int
    height: int
    products: list[LayoutZoneProduct]
